tritium-processor/src/test/java/com/palantir/tritium/examples/DeprecatedType.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.examples;
+
+import com.palantir.tritium.annotations.Instrument;
+
+@Instrument
+@SuppressWarnings("deprecation")
+public interface DeprecatedType {
```

Comment:
Do we care about handling the case where the `@Deprecated` annotation is applied to the type?

Reply:
The `@Deprecated` type case should already be handled & tested:


https://github.com/palantir/tritium/blob/e80ad5b98fbf82115b320c93d09e0f5a83edb1b1/tritium-processor/src/test/java/com/palantir/tritium/examples/DeprecatedInterface.java#L21-L26

https://github.com/palantir/tritium/blob/e80ad5b98fbf82115b320c93d09e0f5a83edb1b1/tritium-processor/src/test/resources/com/palantir/tritium/examples/InstrumentedDeprecatedInterface.java.generated#L13-L15

And the deprecated method case:

https://github.com/palantir/tritium/blob/e80ad5b98fbf82115b320c93d09e0f5a83edb1b1/tritium-processor/src/test/java/com/palantir/tritium/examples/DeprecatedMethod.java#L21-L25

https://github.com/palantir/tritium/blob/e80ad5b98fbf82115b320c93d09e0f5a83edb1b1/tritium-processor/src/test/resources/com/palantir/tritium/examples/InstrumentedDeprecatedMethod.java.generated#L40-L43